jlm/util/Hash.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 Nico Reißmann <nico.reissmann@gmail.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_UTIL_HASH_HPP
+#define JLM_UTIL_HASH_HPP
+
+#include <functional>
+#include <string_view>
+
+namespace jlm::util
+{
+
+/**
+ * Our own version of std::hash that also supports hashing std::pair
+ */
+template<typename T>
+struct Hash : std::hash<T>
```

Comment:
TBH I am never sure which std:: classes you are allowed to derive from and which not. With rare exceptions, things in std:: should be treated as final. I am unclear about std::hash. Is hash-of-pair a recurring thing?

Reply:
There is at least one place where HashSet of pair comes up, and I would think it is a common enough use case that it comes up again.

I'm noticing that the implementation currently does not support pairs of pairs, due to not using `util::Hash` in the implementation.

Regarding not inheriting from std::hash, I tried looking for information about it being illegal or discouraged. Since there is only one method, operator(), we could manually implement it and call `std::hash<T>()(value)` from there.